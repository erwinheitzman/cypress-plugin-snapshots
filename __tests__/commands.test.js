const { before, after, Cypress } = global;
const configMock = require('../src/config');

describe('commands', () => {
  const initConfigSpy = jest.spyOn(configMock, 'initConfig');

  it('initCommands - isHeadless = true', () => {
    const { initCommands } = require('../commands');
    initCommands();

    expect(Cypress.Commands.add).toBeCalled();
    expect(Cypress.Commands.add.mock.calls.length).toEqual(2);
    expect(Cypress.Commands.add.mock.calls[0][0]).toEqual('toMatchSnapshot');
    expect(before).not.toBeCalled(); // isHeadless = true
    expect(after).toBeCalled();
  });

  it('initCommands - isHeadless = false', () => {
    const { initCommands } = require('../commands');
    Cypress.browser.isHeadless = false;
    initCommands();
    expect(before).toBeCalled(); // isHeadless = false
  });

  [
    {
      title: 'empty config',
      config: {}
    },
    {
      title: 'custom config',
      config: {
        autoCleanUp: true,
        excludeFields: ['exclude'],
        imageConfig: {
          createDiffImage: false
        },
        screenshotConfig: {
          blackout: ['blackout']
        }
      }
    }
  ].forEach((item) => {
    it(item.title, () => {
      Cypress.env.mockReturnValueOnce(item.config);
      const { initCommands } = require('../commands');
      initCommands();

      const actual = initConfigSpy.mock.results[0].value;

      expect(actual).toMatchSnapshot();
    });
  });
});
