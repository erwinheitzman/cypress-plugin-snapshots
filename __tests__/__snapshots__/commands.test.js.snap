// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commands custom config 1`] = `
Object {
  "autoCleanUp": true,
  "autopassNewSnapshots": true,
  "backgroundBlend": "difference",
  "diffFormat": "side-by-side",
  "diffLines": 3,
  "excludeFields": Array [
    "exclude",
  ],
  "formatJson": true,
  "ignoreExtraArrayItems": false,
  "ignoreExtraFields": false,
  "imageConfig": Object {
    "createDiffImage": false,
    "failOnSnapshotDiff": true,
    "resizeDevicePixelRatio": true,
    "threshold": 0.1,
    "thresholdType": "percent",
  },
  "normalizeJson": true,
  "prettier": true,
  "prettierConfig": Object {
    "html": Object {
      "endOfLine": "lf",
      "parser": "html",
      "tabWidth": 2,
    },
  },
  "screenshotConfig": Object {
    "blackout": Array [
      "blackout",
      ".snapshot-diff",
    ],
    "log": false,
  },
  "updateSnapshots": false,
}
`;

exports[`commands empty config 1`] = `
Object {
  "autoCleanUp": false,
  "autopassNewSnapshots": true,
  "backgroundBlend": "difference",
  "diffFormat": "side-by-side",
  "diffLines": 3,
  "excludeFields": Array [],
  "formatJson": true,
  "ignoreExtraArrayItems": false,
  "ignoreExtraFields": false,
  "imageConfig": Object {
    "createDiffImage": true,
    "failOnSnapshotDiff": true,
    "resizeDevicePixelRatio": true,
    "threshold": 0.1,
    "thresholdType": "percent",
  },
  "normalizeJson": true,
  "prettier": true,
  "prettierConfig": Object {
    "html": Object {
      "endOfLine": "lf",
      "parser": "html",
      "tabWidth": 2,
    },
  },
  "screenshotConfig": Object {
    "blackout": Array [
      ".snapshot-diff",
    ],
    "log": false,
  },
  "updateSnapshots": false,
}
`;
